backtrace-library/src/main/java/backtraceio/library/metrics/BacktraceEventsHandler.java

```diff
@@ -0,0 +1,202 @@
+package backtraceio.library.metrics;
+
+import android.content.Context;
+import android.os.Handler;
+import android.os.Message;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentLinkedDeque;
+
+import backtraceio.library.common.BacktraceMathHelper;
+import backtraceio.library.common.BacktraceSerializeHelper;
+import backtraceio.library.interfaces.Api;
+import backtraceio.library.logger.BacktraceLogger;
+import backtraceio.library.models.BacktraceMetricsSettings;
+import backtraceio.library.services.BacktraceHandlerThread;
+import backtraceio.library.services.BacktraceReportSender;
+
+public abstract class BacktraceEventsHandler extends Handler {
+
+    private final static transient String LOG_TAG = BacktraceEventsHandler.class.getSimpleName();
+
+    protected final BacktraceHandlerThread backtraceHandlerThread;
+
+    /**
+     * Number of events in the queue
+     */
+    public int getCount() {
+        return events.size();
+    }
+
+    /**
+     * Maximum number of events in store. If number of events in store hit the limit
+     * BacktraceMetrics instance will send data to Backtrace.
+     */
+    private int maxNumEvents = 350;
```

Comment:
Why do we want to use shortcuts? Wouldn't it be better if we will use `maximumNumberOfEvents` instead? Also since it's a const, shouldn't it be upper case?

Reply:
Sounds good, will fix the naming of this constant!

Reply:
This isn't a const, user can change it, we just provide a default. But I did change it to use the full variable name now, thank you @konraddysput!